packages/backend/src/tests/propertyBased/arbitraries/cardinality.ts

```diff
@@ -0,0 +1,109 @@
+import { fc } from '@fast-check/vitest';
+import {
+    AnySchema,
+    getSelectableColumnsFromGenericDbSchema,
+    getWhereableColumnsFromGenericDbSchema,
+    getTableNamesFromGenericDbSchema,
+} from '@synthql/queries';
+
+type Cardinality = 'one' | 'maybe' | 'many';
+
+export function generateQueryArbitrary(
+    schema: AnySchema,
+    allValuesMap: Map<string, Array<any>>,
+    cardinality: Cardinality,
+) {
+    return fc
+        .constantFrom(...getTableNamesFromGenericDbSchema(schema))
+        .chain((table) =>
+            fc.record({
+                from: fc.constant(table),
+                select: fc
+                    .constantFrom(
+                        getSelectableColumnsFromGenericDbSchema(schema, table),
+                    )
+                    .chain((keys) =>
+                        fc.object({
+                            key: fc.constantFrom(
+                                ...getSelectableColumnsFromGenericDbSchema(
+                                    schema,
+                                    table,
+                                ),
+                            ),
+                            values: [fc.boolean()],
+                            maxDepth: 0,
+                            maxKeys: keys.length,
+                        }),
+                    ),
+                where: fc
+                    .constantFrom(
+                        ...getWhereableColumnsFromGenericDbSchema(
+                            schema,
+                            table,
+                        ),
+                    )
+                    .chain((column) =>
+                        fc.object({
+                            key: fc.constant(column),
+                            values: [
+                                whereValueArbitrary(
+                                    allValuesMap,
+                                    table,
+                                    column,
+                                ),
+                            ],
+                            maxDepth: 0,
+                            maxKeys: 1,
+                        }),
+                    ),
+                limit: fc.integer({
+                    min: 1,
+                }),
+                cardinality: fc.constant(cardinality),
+            }),
+        );
+}
+
+export function generateEmptyQueryArbitrary(
```

Comment:
Maybe you misunderstood me when I was taking about empty queries:

I actually meant that if in your db you have a column ID with values 1,2,3 and you search for select * from table where id = $1 and you know that $1 is NOT in the set 1,2,3 then that query should return:

1. many: []
2. one: throw
3. maybe: null